executor/index_merge_reader.go

```diff
@@ -126,27 +124,35 @@ func (e *IndexMergeReaderExecutor) Open(ctx context.Context) error {
 	workCh := make(chan *lookupTableTask, 1)
 	fetchCh := make(chan *lookupTableTask, len(kvRangess))
 
-	e.startIndexMergeProcessWorker(ctx, workCh, fetchCh, len(kvRangess))
-	for i := 0; i < len(kvRangess); i++ {
+	e.startIndexMergeProcessWorker(ctx, workCh, fetchCh)
+
+	var err error
+	for i := 0; i < len(kvRangess) && err == nil; i++ {
 		e.partialWorkerWg.Add(1)
 		if e.indexes[i] != nil {
-			err := e.startPartialIndexWorker(ctx, kvRangess[i], fetchCh, i)
-			if err != nil {
-				return err
-			}
+			err = e.startPartialIndexWorker(ctx, kvRangess[i], fetchCh, i)
 		} else {
 			e.startPartialTableWorker(ctx, fetchCh, i)
 		}
 	}
+	go e.waitPartialWorkersAndCloseFetchChan(fetchCh)
+	if err != nil {
+		return err
+	}
 	e.startIndexMergeTableScanWorker(ctx, workCh)
 	return nil
 }
 
-func (e *IndexMergeReaderExecutor) startIndexMergeProcessWorker(ctx context.Context, workCh chan<- *lookupTableTask, fetch <-chan *lookupTableTask, partialWorkerCount int) {
+func (e *IndexMergeReaderExecutor) waitPartialWorkersAndCloseFetchChan(fetchCh chan *lookupTableTask) {
+	e.partialWorkerWg.Wait()
```

Comment:
1. We do not need to define `partialWorkerWg` as a member attribute of `IndexMergeReaderExecutor`.
2. We do not need to wait for `partialWorkerWg` in  `IndexMergeReaderExecutor.Close`